gravitee-apim-console-webui/src/management/api/api-notification/api-notification.component.ts

```diff
@@ -0,0 +1,210 @@
+  protected notifiers: Notifier[];
+  protected notificationsSummary$: Observable<NotificationSummary[]> = this.notifications$.asObservable().pipe(
+    map((notifications) => {
+      return notifications?.map((notification) => {
+        return {
+          id: notification.id ?? notification.config_type,
+          name: notification.name,
+          subscribedEvents: notification.hooks.length ?? 0,
+          notifier: this.notifiers.find((n) => n.id === notification.notifier),
+          isPortalNotification: notification.config_type === 'PORTAL',
+        };
+      });
+    }),
+    map((notifications) => {
+      if (notifications != null) {
+        this.loading = false;
+        return notifications;
+      }
+      return [];
+    }),
+  );
+
+  constructor(
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly notificationService: ApiNotificationSettingsService,
+    private readonly matDialog: MatDialog,
+    private readonly snackBarService: SnackBarService,
+    private readonly permissionService: GioPermissionService,
+  ) {}
+
+  public ngOnInit(): void {
+    combineLatest([this.notificationService.getNotifiers(this.apiId), this.notificationService.getAll(this.apiId)])
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(([notifiers, notifications]) => {
+        this.notifiers = notifiers;
+        this.notifications$.next(notifications);
+      });
+  }
+
+  public ngOnDestroy(): void {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+    this.notifications$.unsubscribe();
+  }
+
+  protected addNotification() {
+    this.matDialog
+      .open<NotificationAddDialogComponent, NotificationAddDialogData, NotificationAddDialogResult>(NotificationAddDialogComponent, {
+        width: GIO_DIALOG_WIDTH.MEDIUM,
+        data: {
+          notifiers: this.notifiers,
+        },
+        role: 'dialog',
+        id: 'addNotificationDialog',
+      })
+      .afterClosed()
+      .pipe(
+        filter((result) => !!result),
+        switchMap((newNotification) =>
+          this.notificationService.create(this.apiId, {
+            name: newNotification.name,
+            notifier: newNotification.notifierId,
+            referenceType: 'API',
+            referenceId: this.apiId,
+            config_type: 'GENERIC',
+            hooks: [],
+          }),
+        ),
+        tap(() => {
+          this.snackBarService.success('Notification created successfully');
+          this.refreshList();
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  protected deleteNotification(notification: NotificationSummary) {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
+        width: '500px',
```

Comment:
I just checked on ui-particles and GIO_DIALOG_WIDTH small equals 440 and medium 600. Should we use the predefined value here ?,

Reply:
we should 🙈  I will update it